### ISSUE TITLE
More informative error messages for Docker daemon permission issues

### ISSUE THREAD

**Issue by nickjj** [docker/docker#27240]
**Description:**

When trying to pull down an image with a system user that does not belong to the `docker` group without sudo, it fails as expected but the error reported is not very helpful.

**Steps to reproduce the issue:**

1. Install Docker
2. Login to the system with a user not in the docker group
3. docker pull hello-world

**Describe the results you received:**

```
vagrant@pulltest:~$ docker pull hello-world
Using default tag: latest
Warning: failed to get default registry endpoint from daemon (Cannot connect to the Docker daemon. Is the docker daemon running on this host?). Using system default: https://index.docker.io/v1/
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

**Describe the results you expected:**

```
Warning: user `vagrant` does not have permission to access the Docker daemon. Does `vagrant` belong to the `docker` group?
```

**Output of `docker version`:**

```
vagrant@pulltest:~$ docker --version
Docker version 1.12.1, build 23cf638
```

**Output of `docker info`:**

**Note**: Even a `docker info` with fail here too with *Cannot connect to the Docker daemon. Is the docker daemon running on this host?* when the user does not belong to the group.

I have added the vagrant user to the docker group to run `docker info` successfully.

```
vagrant@pulltest:~$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 7
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null overlay host bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options:
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 494.4 MiB
Name: pulltest
ID: PW76:TH7V:BLUW:LKZU:IEJF:DLOY:XR7O:5PEV:WECQ:ESG4:O7WR:VNLM
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

The test was performed in a `debian/contrib-jessie64` Vagrant driven VirtualBox.

**Comment by milindchawre** [comment]
#dibs
I would like to work on it.

**Comment by nickjj** [comment]
@milindchawre If you do, here's the `Vagrantfile` I used for testing:

```ruby
VAGRANTFILE_API_VERSION = '2'

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box = 'debian/contrib-jessie64'
  config.ssh.insert_key = false

  config.vm.provider :virtualbox do |vm|
    vm.name = 'pulltest'
    vm.memory = 512
    vm.cpus = 1
    vm.customize ['modifyvm', :id, '--natdnshostresolver1', 'on']
    vm.customize ['modifyvm', :id, '--ioapic', 'on']
  end

  config.vm.define :pulltest
  config.vm.hostname = 'pulltest'
  config.vm.network :private_network, ip: '192.168.42.42'
end

**Comment by cpuguy83** [comment]
This is already fixed on master by bec07890aad6c7a6699623ab8a61b37e65a7e63d

Thanks!

**Comment by milindchawre** [comment]
I verified this with latest master code and its working as expected.
![dockersnap](https://cloud.githubusercontent.com/assets/21288765/19360308/9a5c6ed0-919b-11e6-92e2-dd5eea3ba57f.PNG)
